### PR TITLE
Select an arbitrary websocket subprotocol proposed by the client

### PR DESCRIPTION
**Pull request by raboof**
Simple (on par with Netty) solution to #7895. Later we could extend our API to
allow the framework user to select a subprotocol instead.

### CODE REVIEW

File: framework/src/play-akka-http-server/src/main/scala/akka/http/play/WebSocketHandler.scala [hunk 1]
```diff
@@ -18,12 +18,23 @@ import play.core.server.common.WebSocketFlowHandler.{ MessageType, RawMessage }
...
 
+  /**
+   * Handle a WebSocket without selecting a subprotocol
+   *
+   * This may cause problems with clients that propose subprotocols in the
+   * upgrade request and expect the server to pick one, such as Chrome.
+   *
+   * See https://github.com/playframework/playframework/issues/7895
+   */
+  def handleWebSocket(upgrade: UpgradeToWebSocket, flow: Flow[Message, Message, _], bufferLimit: Int): HttpResponse =
```
Comment (gmethvin) [inline framework/src/play-akka-http-server/src/main/scala/akka/http/play/WebSocketHandler.scala]: I think we should at least deprecate this. I'm not even sure this class should be considered part of the public API.

**Review by marcospereira — COMMENTED** [review]
Thanks, @raboof.

How hard is to add a test for this in https://github.com/playframework/playframework/blob/master/framework/src/play-integration-test/src/test/scala/play/it/http/websocket/WebSocketSpec.scala?

**Comment by raboof** [comment]
Not really AFAICS, as it seems WebSocketClient doesn't allow setting that header...

**Review by marcospereira — APPROVED** [review]
LGTM!

Thank you, @raboof.

**Comment by marcospereira** [comment]
Backport to 2.7.x: 9d1f9fef5681b622e2f6d50700d63e911379951c

**Comment by ppanda2-ford** [comment]
How to use this feature? When we use WebSocket.accept(...) or WebSocket.acceptOrResult(...) with AkkaHttp Server or the Netty Server, it does not pick the subprotocol. What I am missing here, is there any extra configuration needed for this?

**Comment by sergiuszkierat** [comment]
@marcospereira is any chance for backport to **2.6.x** branch?

**Comment by mkurz** [comment]
With
- #14011 

it's now possible to select the sub protocol on app level.

**Pull request merged**